docs/example_nodejs.md

```diff
@@ -1,34 +1,313 @@
+
+Here's what our timing looks like now:
+
+| Approach | Deploy Time (after initial)
+|---|---|
+| Naive | 11.31-14.21s |
+| Optimized Dockerfile | 3.25-4.12s |
+{:.benchmark-report}
+
+Pretty good! But Tilt has some tricks up its sleeve to make it even faster.
+
+## Step 3: Let's Optimize It EVEN MORE
+
+When we make a change to a file, we currently have to build an image, deploy new Kubernetes configs,
+and wait for Kubernetes to schedule the pod.
+
+With Tilt, we can skip all of these steps, and instead
+[live_update](live_update_tutorial.html) the pod in place.
+
+The first thing we need to do is change how our app is invoked: we're going to run it via [nodemon](https://nodemon.io/), a utility that monitors source files for changes and restarts your app as necessary.  In this branch, as reflected in [package.json](https://github.com/windmilleng/tilt-example-nodejs/blob/master/3-recommended/package.json), we've already run `yarn add --dev nodemon` to add nodemon as a dev dependency.
+
+[Our new Tiltfile](https://github.com/windmilleng/tilt-example-nodejs/blob/master/3-recommended/Tiltfile) contains the following new code:
+
+```python
+# Add a live_update rule to our docker_build
+congrats = "🎉 Congrats, you ran a live_update! 🎉"
+docker_build('example-nodejs-image', '.',
+    build_args={'node_env': 'development'},
+    entrypoint='yarn run nodemon /app/index.js',
+    live_update=[
+        sync('.', '/app'),
+        run('cd /app && yarn install', trigger=['./package.json', './yarn.lock']),
+
+        # if all that changed was start-time.txt, make sure the server
+        # reloads so that it will reflect the new startup time
+        run('touch /app/index.js', trigger='./start-time.txt'),
+
+        # add a congrats message!
+        run('sed -i "s/Hello cats!/{}/g" /app/views/index.mustache'.
+            format(congrats)),
+])
+```
+We've added some new parameters to `docker_build` that tell the container to use nodemon:
+1. the `entrypoint` parameter overrides the `ENTRYPOINT` specified in the Dockerfile; now when the container executes, it will run `yarn run nodemon /app/index.js`
+2. the `build_args` parameter corresponds to this Dockerfile change:
+    ```
+    # Default value; will be overridden by build_args, if passed
+    ARG node_env=production
+
+    ENV NODE_ENV $node_env
+    ```
+   Together, these changes mean that when we build this Dockerfile via this Tiltfile, we set the env var `$NODE_ENV=development`, and our `yarn install` call will install dev dependencies.
+
+With the help of Tilt, we've done a pretty cool thing: we've set it up so this Dockerfile can run the app for development (when invoked via Tilt) or for production (when invoked otherwise)---because we at Tilt think that the closer your dev and prod configs are, the more unpleasant surprises you avoid.
```

Comment:
```suggestion
With the help of Tilt, we've done a pretty cool thing: we've set it up so this Dockerfile can run the app for development (when invoked via Tilt) or for production (when invoked otherwise).
```

Reply:
FINE I'll use less words thhbt. (Good call.)